packages/react/src/components/TextInput/TextInput.js

```diff
@@ -83,8 +91,10 @@ const TextInput = React.forwardRef(function TextInput(
     <div className={helperTextClasses}>{helperText}</div>
   ) : null;
 
+  const { isFluid } = useContext(FormContext);
```

Comment:
Wondering if we can we make this a prop... I imagine usage of context may stem from design guide dictating that fluid is per-form basis. OTOH software designs tend to go further, which is, setting API in different levels (layers). So we can break down "per-form fluid mode" for component implementation, one is a pure mean to change fluid/regular modes (e.g. `<TextInput layout="fluid">`), another is app-level mean to ensure all form items is the desired mode, e.g.:

```javascript
const AppTextInput = props => {
  const { useFluid } = useContext(FormContext);
  return <TextInput {...props} layout={useFluid ? 'fluid' : 'regular'} />;
};
```

Reply:
@asudoh we decided to use context because we wanted to avoid having the user set a `fluid` prop for each individual form element. Since they can only be used in the context of a `FluidForm`, we wanted to make it as easy as possible for users to create a fluid form, while not allowing much deviation from design.

Reply:
I see, thank you for the background @tw15egan. I tend to avoid making "syntactic sugar" thing affect system design (UX-level requirement and system design are often different animals), but I at least now see where it came from.